src/unix/fs.c

```diff
@@ -217,6 +217,88 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  int r;
+#ifdef UV__O_CLOEXEC
+  int (*mkostemp_function)(char*, int);
+  static int no_cloexec_support;
+#endif
+  char *path;
+  int path_length;
+
+  path = (char*) req->path;
```

Comment:
This is a problem already existing with `uv_fs_mkdtemp`, no? Anyway, I made it a `char*` and the tests pass. If you're saying the ABI/API is fine, then we're good to go?

I've also fixed `uv_fs_req_cleanup`.